worldedit-core/src/main/java/com/sk89q/worldedit/command/GenerationCommands.java

```diff
@@ -130,6 +130,74 @@ public int cyl(Actor actor, LocalSession session, EditSession editSession,
         return affected;
     }
 
+    @Command(
+        name = "/cone",
+        desc = "Generates a cone."
+    )
+    @CommandPermissions("worldedit.generation.cone")
+    @Logging(PLACEMENT)
+    public int cone(Actor actor, LocalSession session, EditSession editSession,
+                   @Arg(desc = "The pattern of blocks to generate")
+                       Pattern pattern,
+                   @Arg(desc = "The radii of the cone. 1st is N/S, 2nd is E/W")
+                   @Radii(2)
+                       List<Double> radii,
+                   @Arg(desc = "The height of the cone", def = "1")
+                       int height,
+                   @Switch(name = 'h', desc = "Make a hollow cone")
+                       boolean hollow,
+                    @Arg(desc = "Thickness of the hollow cone", def = "1")
+                        double thickness,
+                    @Arg(desc = "Adjusts the vertical offset of the cone's tip", def = "0")
+                        int tipOffset,
+                    @Arg(desc = "Adjusts the cone's radii by this factor without extending the radii bounds", def = "1,1")
+                    @Radii(2)
+                        List<Double> radiiShape
+    ) throws WorldEditException {
+        double radiusX;
+        double radiusZ;
+        switch (radii.size()) {
+            case 1:
+                radiusX = radiusZ = Math.max(1, radii.get(0));
+                break;
+
+            case 2:
+                radiusX = Math.max(1, radii.get(0));
+                radiusZ = Math.max(1, radii.get(1));
+                break;
+
+            default:
+                actor.printError(TranslatableComponent.of("worldedit.cone.invalid-radius"));
```

Comment:
Needs corresponding entry in the English base translations, `worldedit-core/src/main/resources/lang/strings.json`. Put it next to the other errors of its kind.